dask/backends.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import annotations
+
+import warnings
+from functools import lru_cache
+
+from dask.compatibility import entry_points
+
+
+class DaskBackendIOEntrypoint:
+    """Base Collection-IO Backend Entrypoint Class"""
+
+    @property
+    def fallback(self) -> DaskBackendIOEntrypoint | None:
+        """Fallback entrypoint object.
+
+        Returning anything other than ``None`` requires that
+        ``move_from_fallback`` be properly defined.
+        """
+        return None
+
+    def move_from_fallback(self, x):
+        """Move a Dask collection from the fallback backend"""
+        raise NotImplementedError
+
+    def _get_fallback_attr(self, attr):
+        """Return the fallback version of the specified attribute"""
+        if self.fallback is None:
+            raise ValueError(f"Fallback is not supported for {self}")
+
+        def wrapper(*args, **kwargs):
+            return self.move_from_fallback(
+                getattr(self.fallback, attr)(*args, **kwargs)
+            )
+
+        return wrapper
+
+
+@lru_cache(maxsize=1)
+def detect_entrypoints():
+    entrypoints = entry_points("dask.backends")
+    return {ep.name: ep for ep in entrypoints}
+
+
+class BackendIODispatch:
+    """Simple backend dispatch for collection-IO functions"""
+
+    def __init__(self, name=None):
+        self._lookup = {}
+        if name:
+            self.__name__ = name
+
+    def register_backend(self, backend: str, cls_target: DaskBackendIOEntrypoint):
+        """Register a target class for a specific backend label"""
+
+        def wrapper(cls_target):
+            if isinstance(backend, tuple):
+                for b in backend:
+                    self.register_backend(b, cls_target)
```

Comment:
If the type annotation of `backend` is correct (maybe it's not) then I don't think we'll ever go inside this `if` block. IIUC the intent of this `if` is to support things like `register_backend(("foo", "bar". "baz"), MyCustomBackend)`. This doesn't seem unreasonable, but also doesn't seem necessary. I'd prefer to keep the initial API simple (i.e. `backend` must be a string) and then extend and make things more flexible if we find the simple thing cumbersome.

Reply:
I agree - removed.